spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/convert/EntityPropertyValueProvider.java

```diff
@@ -33,8 +36,19 @@
 public class EntityPropertyValueProvider implements PropertyValueProvider<DatastorePersistentProperty> {
 	private Entity entity;
 
-	EntityPropertyValueProvider(Entity entity) {
+	private ConversionService conversionService;
+
+	private ConversionService internalConversionService;
+
+	private Function<Class<?>, DefaultDatastoreEntityConverter.TwoStepConversion> conversionGetter;
```

Comment:
A reference to a function here seems odd. I think we need to think a bit more about this design.